evalml/pipelines/utils.py

```diff
@@ -265,7 +270,9 @@ def _make_component_list_from_actions(actions):
         List of components used to address the input actions
     """
     components = []
-    for action in actions:
-        if action.action_code == DataCheckActionCode.DROP_COL:
-            components.append(DropColumns(columns=action.metadata["columns"]))
+    cols_to_drop = []
+    for drop_action in [action for action in actions if action.action_code == DataCheckActionCode.DROP_COL]:
+        cols_to_drop.extend(drop_action.metadata["columns"])
```

Comment:
Looks good. What happens if `drop_action.metadata` has no key "columns"? Do we have test coverage for that case? I suggest you use `drop_action.metadata.get("columns", [])`